### ISSUE TITLE
controller tests all broken on rails 5 because they want to be integration tests

### ISSUE THREAD

**Issue by grosser** [blowmage/minitest-rails#195]
Did not see the memo that controller tests are bad now ... seem to work fine when used manually via `< ActionController::TestCase` ... but `describe XyZController` style no longer works :(

**Comment by grosser** [comment]
using this to fix it

```Ruby
# https://github.com/blowmage/minitest-rails/issues/195
class ActionController::TestCase
  # Use AD::IntegrationTest for the base class when describing a controller
  register_spec_type(self) do |desc|
    Class === desc && desc < ActionController::Metal
  end
end

```

**Comment by blowmage** [comment]
@grosser See #194.

**Comment by grosser** [comment]
Saw that ... but I don't want it to be optional ... I'm thinking it should
just work since controller tests are still supported / not discouraged ?

**Comment by blowmage** [comment]
Use of ActionController::TestCase _is_ discouraged. The generators now produce test files that inherit from ActionDispatch::IntegrationTest. AC::TC is deprecated in Rails 5.0 and will be removed in Rails 5.1.

The problem we face is what is the behavior of the spec DSL when describing a Controller class. Do we follow the Rails 5.0 approach and use AD::ID? If we do this we need a way for folks to opt-in for the Rais 4 approach. Or do we follow the Rails 4 approach and use AC::TC? If we do this we need a way for folks to opt in for the Rails 5.0 approach

The problem with following the Rails 4 approach is that it changes again once Rails 5.1 comes out. Besides being backwards looking it means we have to support two different opt-in options in Rails 5.0 and 5.1.

It is cleaner IMO to have minitest-rails 3.0 embrace the new way of handling controller tests, and have a single opt-in mechanism for specifying the old behavior. That mechanism, looking for the rails-controller-testing gem, is also how AC::TC will be made available in Rails 5.1.

**Comment by grosser** [comment]
Nvm then ... did not see any deprecations :(
Should use the new style to be compatible going forward ... I'll use the hack in the meantime to do our upgrades more slowly / 1-by-1
